### PR TITLE
LPS-162047 Tag Facet ordering

### PR DESCRIPTION
**Pull request by felipelorenz**
https://issues.liferay.com/browse/LPS-162047

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/tag/facet/configuration.jsp [hunk 1]
```diff
@@ -78,6 +78,14 @@ TagFacetPortletPreferences tagFacetPortletPreferences = new TagFacetPortletPrefe
 
 				<aui:input label="frequency-threshold" name="<%= PortletPreferencesJspUtil.getInputName(TagFacetPortletPreferences.PREFERENCE_KEY_FREQUENCY_THRESHOLD) %>" value="<%= tagFacetPortletPreferences.getFrequencyThreshold() %>" />
 
+				<aui:select label="order-terms-by" name="<%= PortletPreferencesJspUtil.getInputName(TagFacetPortletPreferences.PREFERENCE_KEY_ORDER) %>" value="<%= tagFacetPortletPreferences.getOrder() %>">
+					<aui:option label="" value="" />
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/tag/facet/configuration.jsp:82]: @felipelorenz Could you remove that blank option and put "term-frequency-descending" as the default option to be displayed?

Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/tag/facet/configuration.jsp:82]: @felipelorenz Could you remove that blank option and put "term-frequency-descending" as the default option to be displayed?

File: modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java [hunk 2]
```diff
@@ -181,8 +183,101 @@ public void testOneTermWithPreviousSelection() throws Exception {
...
+
+		List<BucketDisplayContext> bucketDisplayContexts =
+			assetTagsSearchFacetDisplayContext.getBucketDisplayContexts();
+
+		String nameFrequencyString = _buildNameFrequencyString(
+			bucketDisplayContexts);
+
+		Assert.assertEquals(
+			bucketDisplayContexts.toString(),
+			"alpha:4|bravo:5|charlie:5|delta:6", nameFrequencyString);
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java:205]: @felipelorenz That test must prove to us that the terms are ordered by Frequency Ascending, I think you must have term names as an expected result that are not ordered as Value Ascending.

File: modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java [hunk 3]
```diff
@@ -181,8 +183,101 @@ public void testOneTermWithPreviousSelection() throws Exception {
 			assetTagsSearchFacetDisplayContext.isRenderNothing());
 	}
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java:185]: @felipelorenz @joshuacords  Maybe could be a good idea to add two different asserts in each test, the first one will be to test the ordering without any equal frequency or equal term name, and the second one will be like the tests are already asserting, what you guys think about that? 

That way we will have more consistent tests

**Review by gustavolimav — CHANGES_REQUESTED** [review]

File: modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java (hunk 3 above)
Comment (joshuacords) [reply modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/AssetTagsSearchFacetDisplayContextTest.java:185]: Yes, we should be testing the defaults as well - great point Gustavo!

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

**Review by gustavolimav — APPROVED** [review]

**Comment by gustavolimav** [comment]
Good to go!

**Comment by gustavolimav** [comment]
ci:forward

**Pull request closed without merging**